### PR TITLE
Adds a bunch of custom design disks, sorts some design files

### PR DESCRIPTION
**Pull request by thgvr**
## About The Pull Request
Adds prefab design disks for future mapping/loot usage including:
* Surgical tools
* Every tier of cybernetic organ
* IPC components
* Chemistry equipment (pill bottles, spray bottles, xlarge beaker, reagent analyzer, iv bags)

Mostly medical stuff for now. I am taking suggestions for more things that would be cool on disks so I can eventually move a lot of stuff out of protolathe lists and R&D in general.

## Why It's Good For The Game
More things to be used as useful loot is good, expanding our autolathe design disk feature.

## Changelog

:cl:
add: Prefab design disks for chemistry equipment, surgical tools, cyber organs, and IPC components.
balance: Buffed normal mining satchel from 50 total items to 150 total items.
balance: Buffed RPED from 50 total items to 100 total items.
imgadd: Changed the default TGUI background icon to a Shiptest one.
/:cl:

### CODE REVIEW

**Comment by Zevotech** [comment]
Inducers, Super/Hyper Cells, RCD/RPD disks and Experimental construction tools disks. You could even add disks that FINALLY allow you to print syndicate tools and inducers.
Survey tools, Cybernetics (like HUD implants), disks for those sciency mining things that the golems get, and maybe a bluespace mining bag (only requires gold and uranium, so its printable at the autolathe)
Above all else though, disks that let you print anomaly neutralizers would be amazing.

**Comment by thgvr** [comment]
Syndicate tool disk is a really good idea
I might make normal mining satchels less shit. I don't like bluespace stuff because its infinite and lame

**Comment by axelzonvolt** [comment]
anomaly neutralizer, penlite defibrillator frame, gps...

even just the basic stuff that comes with protolathe+console could be put on a disk and found. (basic ammo magazines and  boxes)

**Comment by Zevotech** [comment]
> Syndicate tool disk is a really good idea I might make normal mining satchels less shit. I don't like bluespace stuff because its infinite and lame

something i wanted to suggest but wasnt sure about shoving it on you was a middle tier upgrade. Not quite infinite but also not limited to a mere 50 ore like the base mining satchel.

**Comment by thgvr** [comment]
most of the current bags are pretty powerful, i think mining satchel is actually equal to them, but doesn't account for ores being in high stacks or whatever. I bet I could just raise it to like 150, im pretty sure they can only hold 50 ore or something

**Comment by Zevotech** [comment]
> most of the current bags are pretty powerful, i think mining satchel is actually equal to them, but doesn't account for ores being in high stacks or whatever. I bet I could just raise it to like 150, im pretty sure they can only hold 50 ore or something

yep, only holds around 50 ore but doesnt account for stacking. raising it would be nice because we are pretty exploration and mining focused.

File: code/modules/research/design_disks.dm [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+	custom_materials = list(/datum/material/iron =300, /datum/material/glass =100)
+	var/list/blueprints = list()
+	var/max_blueprints = 1
+
+/obj/item/disk/design_disk/Initialize()
+	. = ..()
+	pixel_x = base_pixel_x + rand(-5, 5)
+	pixel_y = base_pixel_y + rand(-5, 5)
+	for(var/i in 1 to max_blueprints)
+		blueprints += null
```
Comment (Kapu1178) [inline code/modules/research/design_disks.dm]:
```suggestion
	var/list/blueprints
	var/max_blueprints = 1

/obj/item/disk/design_disk/Initialize()
	. = ..()
	pixel_x = base_pixel_x + rand(-5, 5)
	pixel_y = base_pixel_y + rand(-5, 5)
	blueprints = new(max_blueprints)
```

File: code/game/machinery/autolathe.dm [hunk 2]
```diff
@@ -344,7 +352,8 @@
 		for(var/i=1, i<=multiplier, i++)
 			var/obj/item/new_item = new being_built.build_path(A)
 			new_item.autolathe_crafted(src)
-
+			d_disk.used_charges += 1
+			d_disk.check_charges()
```
Comment (MarkSuckerberg) [inline code/game/machinery/autolathe.dm]: probably will work. maybe. I should probably test this but idk
```suggestion
			if(being_built in d_disk.blueprints)
				d_disk.used_charges += 1
				d_disk.check_charges()
```

Comment (Erikafox) [reply code/game/machinery/autolathe.dm]: we have a perfectly good playerbase to test on

Comment (MarkSuckerberg) [inline code/game/machinery/autolathe.dm]:
```suggestion
			if(locate(being_built.type) in d_disk.blueprints)
				d_disk.used_charges += 1
				d_disk.check_charges()
``` attempt 2, I hate this, please think of a better way to do this :pleading_face:

**Comment by thgvr** [comment]
Will finish this up soon

**Comment by thgvr** [comment]
i forgor

File: code/modules/research/design_disks/engineering.dm [hunk 3]
```diff
@@ -0,0 +1,110 @@
+/*
+	This file contains disks with engineering designs.
+*/
+
+// ENGINEERING EQUIPMENT / PARTS
+
+/obj/item/disk/design_disk/engineering/ //Engineering parent type
```
Comment (MarkSuckerberg) [inline code/modules/research/design_disks/engineering.dm:7]:
```suggestion
/obj/item/disk/design_disk/engineering //Engineering parent type
```

File: code/modules/research/design_disks/engineering.dm [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+	name = "design disk - Engineering"
+	desc = "A design disk containing engineering equipment."
+	color = "#eece17"
+	illustration = "hammer"
+	max_blueprints = 1
+
+/obj/item/disk/design_disk/engineering/Initialize() //wip- not sure what else to put here
+	. = ..()
+	blueprints[1] = new /datum/design/inducer
```
Comment (MarkSuckerberg) [inline code/modules/research/design_disks/engineering.dm:16]: slightly advanced welder, t-ray, maybe even a hardhat? so tempted to just say remove all the designs and move over to this system but that's out of scope

File: code/modules/research/design_disks/firearms.dm [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+/obj/item/disk/design_disk/limited/firearm/disposable_gun
+	name = "design disk - disposable guns"
+	desc = "A design disk containing designs for a cheap and disposable gun."
+	illustration = "gun"
+	max_blueprints = 1
+	max_charges = 20
+
+/obj/item/disk/design_disk/limited/firearm/disposable_gun/Initialize()
+	. = ..()
+	blueprints[1] = new /datum/design/disposable_gun/
```
Comment (MarkSuckerberg) [inline code/modules/research/design_disks/firearms.dm:55]:
```suggestion
	blueprints[1] = new /datum/design/disposable_gun
```

**Review by MarkSuckerberg — CHANGES_REQUESTED** [review]
I still think it might be best to move populating designs on disks to a separate proc that gets called in the base class init, but it's up to you if you wanna wait on that

**Comment by tmtmtl30** [comment]
there are some flaws with the way print limitations are implemented. i mean to get to them, but it's not right as-is (you might be able to print after the disk runs out before the TGUI updates, and similarly you can print with multipliers even if there's only one charge remaining on the disk)

**Comment by thgvr** [comment]
More I think about this more I hate the implementation. I have bigger plans

**Comment by MarkSuckerberg** [comment]
just realised you closed this :samsungsob: what are your plans and can they be accomplished in a reasonable amount of time

**Pull request closed without merging**